resources/s3-buckets.go

```diff
@@ -74,6 +76,35 @@ func (e *S3Bucket) Remove() error {
 	return nil
 }
 
+func (e *S3Bucket) retrieveTags() ([]*s3.Tag, error) {
+	input := &s3.GetBucketTaggingInput{
+		Bucket: aws.String(e.name),
+	}
+
+	result, err := e.svc.GetBucketTagging(input)
+	if err != nil {
+		return nil, err
+	}
+
+	return result.TagSet, nil
+}
+
+func (e *S3Bucket) Properties() types.Properties {
+	properties := types.NewProperties()
+	properties.Set("Name", e.name)
+
+	tags, err := e.retrieveTags()
```

Comment:
The `Properties()` function gets called quite often during the whole process. Therefore I suggest to move the `retrieveTags` to `DescribeS3Buckets`.